core/src/main/kotlin/com/alecstrong/sql/psi/core/postgresql/PostgreSql.bnf

```diff
@@ -106,7 +106,7 @@ big_serial_data_type ::= 'BIGSERIAL' | 'SERIAL8'
 
 string_data_type ::= ((( 'CHARACTER' 'VARYING' ) | 'VARCHAR' | 'CHARACTER' | 'CHAR') [ '(' {signed_number} ')' ]) | 'TEXT' | 'UUID' | 'CITEXT'
 
-date_data_type ::= 'DATE' | (('TIME' | 'TIMESTAMP') [ '(' {signed_number} ')' ])
+date_data_type ::= 'DATE' | (('TIME' | 'TIMESTAMP' | 'TIMESTAMPTZ' | ('TIMESTAMP' 'WITH' 'TIMEZONE')) [ '(' {signed_number} ')' ])
```

Comment:
Don't think this is quite correct based on https://www.postgresql.org/docs/9.1/datatype-datetime.html.

For one, the `WITH TIME ZONE` (you're missing the space) and `WITHOUT TIME ZONE` are modifiers on both `TIME` and `TIMESTAMP` and occur _after_ the parenthetical precision expression.

So I think you want something like:
```
'DATE' | (('TIME' | 'TIMESTAMP') [ '(' {signed_number} ')] ] [ ('WITH' | 'WITHOUT') 'TIME' 'ZONE') ] ) | 'TIMESTAMPTZ'
```
with more test cases in your sample
```
some_date DATE,

some_time TIME,
some_time_precision TIME (3),
some_time_without_tz TIME WITHOUT TIME ZONE,
some_time_precision_without_tz TIME (3) WITHOUT TIMEZONE,
some_time_with_tz TIME WITH TIME ZONE,
some_time_precision_with_tz TIME (3) WITH TIMEZONE,

some_timestamp TIMESTAMP,
some_timestamp_precision TIMESTAMP (3),
some_timestamp_without_tz TIMESTAMP WITHOUT TIME ZONE,
some_timestamp_precision_without_tz TIMESTAMP (3) WITHOUT TIMEZONE,
some_timestamp_with_tz TIMESTAMP WITH TIME ZONE,
some_timestamp_precision_with_tz TIMESTAMP (3) WITH TIMEZONE,
```

Reply:
Ah, I see. Sorry, new to bnf.

Reply:
> The SQL standard requires that writing just timestamp be equivalent to timestamp without time zone, and PostgreSQL honors that behavior. 

We also have to cover this use case.